src/components/form-editor/modals/new-question.tsx

```diff
@@ -0,0 +1,397 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  Button,
+  ComboBox,
+  ComposedModal,
+  Form,
+  FormGroup,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Row,
+  Select,
+  SelectItem,
+  TextInput,
+} from "carbon-components-react";
+import { Answer, Concept, ConceptMapping, Question } from "../../../api/types";
+import { Add } from "@carbon/icons-react/next";
+import { SchemaContext } from "../../../context/context";
+import { showToast } from "@openmrs/esm-framework";
+import { useConcepts } from "../../../api/concept";
+import styles from "./modals.scss";
+
+interface CreateQuestionModalProps {
+  questions: any;
+}
+
+const CreateQuestion: React.FC<CreateQuestionModalProps> = ({ questions }) => {
+  const { t } = useTranslation();
+  const { concepts } = useConcepts();
+  const { schema, setSchema } = useContext(SchemaContext);
+  const [openCreateQuestionModal, setOpenCreateQuestionModal] = useState(false);
+  const [questionLabel, setQuestionLabel] = useState("");
+  const [questionType, setQuestionType] = useState("");
+  const [questionId, setQuestionId] = useState("");
+  // Question Options
+  const [renderElement, setRenderElement] = useState("");
+  const [answers, setAnswers] = useState<Answer[]>([]);
+  const [max, setMax] = useState("");
+  const [min, setMin] = useState("");
+  const [concept, setConcept] = useState<string>(null);
+  const [conceptMappings, setConceptMappings] = useState<ConceptMapping[]>([]);
+  const [weekList, setWeekList] = useState<any>([]);
+  const [attributeType, setAttributeType] = useState("");
+  const [rows, setRows] = useState("");
+  const [orderSettingUuid, setOrderSettingUuid] = useState("");
+  const [orderType, setOrderType] = useState("");
+  const [selectableOrders, setSelectableOrders] = useState<Answer[]>([]);
+
+  const types = [
```

Comment:
Just declare this value as a runtime config, (see [example](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-medications-app/src/config-schema.ts#L17-L21)) 

Now you can reference it like:

```javascript

const { types } = useConfig(...);

```

@kumuditha-udayanga Well a better approach is say you have the `baseTypes` which are statically defined above and then have `supportTypesOverrides` which are driven by the configuration approach. Now values within the `supportTypesOverrides` should be able to override the base. I would prefer the latter.

Reply:
Thank you for clarifying this @samuelmale. I made the changes as suggested. Now the render elements and types can be managed through a config.
<img width="1440" alt="Screenshot 2022-09-13 at 1 09 32 AM" src="https://user-images.githubusercontent.com/27630091/189741098-a5410e77-1f26-4486-855f-5fbbd272dfb4.png">